cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -411,6 +433,34 @@ public boolean updateAppAllowed(ApplicationId appId) throws Exception {
     return version.endsWith(ApplicationId.DEFAULT_VERSION);
   }
 
+  /**
+   * To determine whether the app is allowed to be deployed:
+   * Do not deploy when the parent version is not the latest.
+   *
+   * @param appId the id of the application to be determined
+   * @param parentVersion the version of the application from which the app is deployed
+   * @return whether the app version is allowed to be deployed
+   */
+  public boolean deployAppAllowed(ApplicationId appId, String parentVersion) throws Exception {
+    accessEnforcer.enforce(appId, authenticationContext.getPrincipal(), StandardPermission.GET);
+    ApplicationMeta latest = store.getLatestAppVersion(appId);
+    // App does not exist
+    if (parentVersion == null || parentVersion.isEmpty() || latest == null || latest.getSpec() == null) {
+      //parent version in request is null then Allow to create a new one (Cannot create an app
```

Comment:
this comment is inaccurate, is it supposed to say `parent version can only be null if the app does not exist`

Reply:
parent version can also be null (not provided by the user) and the app can still exist and we allow deployment in this case, since this parent version is an optional field

Reply:
I think the comment can be `parent version should be null when the app does not exist`